pkg/api/v1/field_patterns.go

```diff
@@ -0,0 +1,44 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+
+package v1
+
+import (
+	"github.com/go-openapi/errors"
+	"github.com/go-openapi/validate"
+)
+
+// NO TESTS
+
+// FieldPattern field pattern
+// swagger:model FieldPattern
+type FieldPattern struct {
+
+	// regex pattern
+	Pattern string `json:"pattern"`
+
+	// translated msg presented to user
+	Message string `json:"message"`
+}
+
+// FieldPatternWordNumberDashOnly letter, underscore, number and dash
+var FieldPatternWordNumberDashOnly = FieldPattern{
```

Comment:
I'm not sure this name is entirely helpful. `FieldPatternWordNumberDashOnly` and `FieldPatternLetterNumberDashOnly` differ at `Letter` and `Word, where the actual difference is one accepts underscores and the other doesn't.

Reply:
I agree this is somehow confusing. The LetterNumberDashOnly one is only used in EventDriver. The rest of model name will use WordNumberDashOnly. What do you think if I change them to `FieldPatternName` and `FieldPatternNameNoUnderscore`?

Reply:
These seem reasonable.

Reply:
Thanks, will do!